packages/dev/core/src/Meshes/mesh.ts

```diff
@@ -3093,107 +3031,126 @@ export class Mesh extends AbstractMesh implements IGetSetVerticesData {
 
             // Store same normals for every vertex
             for (let localIndex = 0; localIndex < 3; localIndex++) {
-                normals.push(normal.x);
-                normals.push(normal.y);
-                normals.push(normal.z);
+                normals[normalsCount++] = normal.x;
+                normals[normalsCount++] = normal.y;
+                normals[normalsCount++] = normal.z;
             }
         }
 
-        this.setIndices(indices);
-        this.setVerticesData(VertexBuffer.NormalKind, normals, updatableNormals);
+        return normals;
+    }
 
-        // Updating vertex buffers
-        for (kindIndex = 0; kindIndex < kinds.length; kindIndex++) {
-            kind = kinds[kindIndex];
+    /**
+     * @internal
+     * @param flattenNormals 
+     * @returns the Mesh.
+     */
+    private _convertToUnIndexedMesh(flattenNormals = false, updateIndices = true): Mesh {
+        const kinds = this.getVerticesDataKinds();
+        const indices = this.getIndices()!;
+        const data: { [kind: string]: FloatArray } = {};
 
-            if (!newdata[kind]) {
-                continue;
+        const getDuplicatedVertices = (data: FloatArray, stride: number): Float32Array => {

```

Comment:
nit: How about this?
```suggestion
        const separateVertices = (data: FloatArray, stride: number): Float32Array => {
```